### PR TITLE
Parallelize make to reduce build time

### PR DESCRIPTION
**Pull request by naveentatikonda**
### Description
Parallelize `make` to reduce the build time when we are building the plugin or the JNI libraries for the first time. 

### Check List
- [ ] New functionality includes testing.
- [x] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by naveentatikonda** [comment]
Running into some issues while detecting the processor count and passing it to gradle on mac and windows from github workflow. So, hardcoding them for now based on the github runners configuration
https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -117,15 +117,16 @@ jobs:
...
           export FC=/usr/local/Cellar/gcc/12.2.0/bin/gfortran
 
+      # TODO: Detect processor count and set the value of nproc.count
       - name: Run build
         run: |
           if sysctl -n machdep.cpu.features machdep.cpu.leaf7_features | grep -i AVX2
           then
               echo "avx2 available on system"
-              ./gradlew build 
+              ./gradlew build -Dnproc.count=3
```
Comment (navneet1v) [inline .github/workflows/CI.yml:126]: why hardcoded 3?

**Review by navneet1v — COMMENTED** [review]
anything that speeds up build time is great. thanks for adding this feature.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (navneet1v) [reply .github/workflows/CI.yml:126]: got my ans here: https://github.com/opensearch-project/k-NN/pull/2006#issuecomment-2313124924

**Review by navneet1v — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -18,6 +18,7 @@ buildscript {
         opensearch_group = "org.opensearch"
         isSnapshot = "true" == System.getProperty("build.snapshot", "true")
         simd_enabled = System.getProperty("simd.enabled", "true")
+        nproc_count = System.getProperty("nproc.count", "1")
```
Comment (jmazanec15) [inline build.gradle:21]: Can we default to 4?

**Review by shatejas — APPROVED** [review]

**Review by jmazanec15 — APPROVED** [review]

**Pull request merged**